vars/goBuildAndRelease.groovy

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/groovy
+
+def call(body) {
+    def config = [:]
+    body.resolveStrategy = Closure.DELEGATE_FIRST
+    body.delegate = config
+    body()
+
+    toolsNode(toolsImage: 'stakater/pipeline-tools:1.6.0') {
+        container(name: 'tools') {
+            withCurrentRepo(type: 'go') { def repoUrl, def repoName, def repoOwner, def repoBranch ->
+                String chartPackageName = ""
+                String srcDir = WORKSPACE
+                def kubernetesDir = WORKSPACE + "/deployments/kubernetes"
+
+                def chartTemplatesDir = kubernetesDir + "/templates/chart"
+                def chartDir = kubernetesDir + "/chart"
+                def manifestsDir = kubernetesDir + "/manifests"
+
+                def dockerContextDir = WORKSPACE + "/build/package"
+                def dockerImage = repoOwner.toLowerCase() + "/" + repoName.toLowerCase()
+                def dockerImageVersion = ""
+
+                // Slack variables
+                def slackChannel = "${env.SLACK_CHANNEL}"
+                def slackWebHookURL = "${env.SLACK_WEBHOOK_URL}"
+
+                def utils = new io.fabric8.Utils()
+                def git = new io.stakater.vc.Git()
+                def helm = new io.stakater.charts.Helm()
+                def templates = new io.stakater.charts.Templates()
+                def common = new io.stakater.Common()
+                def chartManager = new io.stakater.charts.ChartManager()
+                def docker = new io.stakater.containers.Docker()
+                def stakaterCommands = new io.stakater.StakaterCommands()
+                def slack = new io.stakater.notifications.Slack()
+                try {
+                    stage('Download Dependencies') {
+                        sh """
+                            cd ${srcDir}
+                            make install
+                        """
+                    }
+
+                    stage('Run Tests') {
+                        sh """
+                            cd ${srcDir}
+                            make test
+                        """
+                    }
+
+                    stage('Build Binary') {
+                        sh """
+                            cd ${srcDir}
+                            go build -o ./build/package/${repoName.toLowerCase()}
```

Comment:
Is there a target to build in the makefile? this should be replace by that target